### PR TITLE
tsp, fix subclass of class with additionalProperties - part1

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
code https://github.com/Azure/autorest.java/pull/2499/commits/7833aa3b1ffb0ef46f5034af8c1d0d7ab484c532
Jackson should be good.

pending part2 on fix to stream-style https://github.com/Azure/autorest.java/issues/2500

### CODE REVIEW

File: typespec-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -2062,8 +2062,16 @@ export class CodeModelBuilder {
       }
     } else if (isRecordModelType(this.program, type)) {
       // "pure" Record processed elsewhere
+
       // "mixed" Record that have properties, treat the model as "additionalProperties"
-      const parentSchema = this.processDictionarySchema(type, this.getName(type));
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:2066]: Bug was that previously this `type` get cached as a `DictionarySchema` in this line.

It should be cached as an `ObjectSchema`, while its sourceModel be potentially `DictionarySchema`

**Comment by weidongxu-microsoft** [comment]
As typespec, the `additionalProperty` is not a "meta concept".

It is currently done like this
```ts
model SuperClass is Record<unknown> {
  prop1InSuperClass: string;
}

model Class extends SuperClass {
  prop1InClass: string;
}
```

So it is a basic "extends" syntax.

PS: typespec is discussing writing this as
```ts
model SuperClass {
  ...Record<unknown>;
  prop1InSuperClass: string;
}

model Class extends SuperClass {
  prop1InClass: string;
}
```

File: typespec-tests/src/main/java/com/type/property/additionalproperties/models/IsFloatAdditionalProperties.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public final class IsFloatAdditionalProperties implements JsonSerializable<IsFlo
     private final double id;
 
     /*
-     * The model is from Record<float32> type.
+     * Dictionary of ExtendsUnknownAdditionalPropertiesDerivedAge
      */
```
Comment (haolingdong-msft) [inline typespec-tests/src/main/java/com/type/property/additionalproperties/models/IsFloatAdditionalProperties.java:30]: A bit confused, the dictionay is `Map<String, Double>`, why is this 'Dictionary of `ExtendsUnknownAdditionalPropertiesDerivedAge`', not dictionay of `Double`? Is it saying it's the `age` property of `ExtendsUnknownAdditionalPropertiesDerived` model?

Comment (weidongxu-microsoft) [reply typespec-tests/src/main/java/com/type/property/additionalproperties/models/IsFloatAdditionalProperties.java:30]: Yeah, this is a bit strange. (seems from the doc of the Record<>)

I will take a look.

Comment (weidongxu-microsoft) [reply typespec-tests/src/main/java/com/type/property/additionalproperties/models/IsFloatAdditionalProperties.java:30]: The cause is the `float` got names as `ExtendsUnknownAdditionalPropertiesDerivedAge`. This alone won't affect anything, as we don't generate a float class/struct.

But the `namer` module would find that there is no `@doc` on the Record, and use the name above to coin a description, hence the strange name.

Comment (weidongxu-microsoft) [reply typespec-tests/src/main/java/com/type/property/additionalproperties/models/IsFloatAdditionalProperties.java:30]: fix in https://github.com/Azure/autorest.java/pull/2499/commits/51f4ea5cd0c3869882c1c41f42217d39cd312dcc and https://github.com/Azure/autorest.java/pull/2499/commits/7f5c7434f7dc22cba074a521aeee00df2dc938ee

also disabled `namer` in test, and hide it in readme.

**Review by haolingdong-msft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**